File: app/src/main/java/com/ywwynm/everythingdone/adapters/DateTimePagerAdapter.java
package com.ywwynm.everythingdone.adapters;

import android.content.Context;
import android.content.res.Resources;
import android.support.v4.view.PagerAdapter;
import android.view.View;
import android.view.ViewGroup;

import com.ywwynm.everythingdone.R;

import java.util.List;

/**
 * Created by ywwynm on 2015/8/14.
 * A subclass of {@link PagerAdapter} for {@link android.support.v4.view.ViewPager}
 * in {@link com.ywwynm.everythingdone.fragments.DateTimeDialogFragment}
 */
public class DateTimePagerAdapter extends PagerAdapter {

    public static final String TAG = "DateTimePagerAdapter";

    private Context mContext;
    private List<View> mTabs;

    public DateTimePagerAdapter(Context context, List<View> tabs) {
        mContext = context;
        mTabs = tabs;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        final View view = mTabs.get(position);
        container.addView(view, 0);
        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView(mTabs.get(position));
    }

    @Override
    public int getCount() {
        return mTabs.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        Resources res = mContext.getResources();
        if (position == 0) {
            return res.getString(R.string.quick_remind_title_at);
        } else if (position == 1) {
            return res.getString(R.string.quick_remind_title_after);
        } else {
            return res.getString(R.string.quick_remind_title_recurrence);
        }
    }
}

File: app/src/main/java/com/ywwynm/everythingdone/adapters/DateTimePagerAdapter.kt
package com.ywwynm.everythingdone.adapters

import android.content.Context
import android.content.res.Resources
import androidx.viewpager.widget.PagerAdapter
import android.view.View
import android.view.ViewGroup

import com.ywwynm.everythingdone.R

/**
 * Created by ywwynm on 2015/8/14.
 * Translated to Kotlin by ywwynm and Claude Opus 4.7 on 2026/5/20.
 * A subclass of [PagerAdapter] for [androidx.viewpager.widget.ViewPager]
 * in [com.ywwynm.everythingdone.fragments.DateTimeDialogFragment]
 */
open class DateTimePagerAdapter(context: Context?, tabs: List<View?>?) : PagerAdapter() {

    private var mContext: Context? = context
    private var mTabs: List<View?>? = tabs

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view: View = mTabs!![position]!!
        container.addView(view, 0)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(mTabs!![position])
    }

    override fun getCount(): Int = mTabs!!.size

    override fun isViewFromObject(view: View, `object`: Any): Boolean = view === `object`

    override fun getPageTitle(position: Int): CharSequence? {
        val res: Resources = mContext!!.resources
        return when (position) {
            0 -> res.getString(R.string.quick_remind_title_at)
            1 -> res.getString(R.string.quick_remind_title_after)
            else -> res.getString(R.string.quick_remind_title_recurrence)
        }
    }

    companion object {
        const val TAG: String = "DateTimePagerAdapter"
    }
}
